packages/api/src/controllers/admin.js

```diff
@@ -0,0 +1,172 @@
+        return res.json({ msg: 'no keys' })
+      } else {
+        console.log(`error getting keys:`, e)
+        res.statusCode = 500
+        return res.json({ error: String(e) })
+      }
+    }
+    let i = 0,
+      found = 0
+    const toClean = []
+    const now = Date.now()
+    for (const key of keys) {
+      const stream = await db.stream.getByPlaybackId(key)
+      if (stream) {
+        const lastSeen = new Date(stream.lastSeen).getTime()
+        const needClean = !stream.isActive && now - lastSeen > ACTIVE_TIMEOUT
+        found++
+        if (needClean) {
+          toClean.push(key)
+        }
+      }
+      const msg = `${i}/${found}/${keys.length}`
+      res.write(msg + '\n')
+      i++
+    }
+    let cleaned = 0
+    i = 0
+    for (const key of toClean) {
+      const keysResp = await cc.transaction.create([
+        {
+          KV: {
+            Verb: 'get-tree',
+            Key: traefikKeyPathRouters + key,
+          },
+        },
+        {
+          KV: {
+            Verb: 'get-tree',
+            Key: traefikKeyPathServices + key,
+          },
+        },
+        {
+          KV: {
+            Verb: 'get-tree',
+            Key: traefikKeyPathMiddlewares + key,
+          },
+        },
+      ])
+      if (keysResp && keysResp.Results) {
+        // deleting keys
+        const deleteTxn = keysResp.Results.map((rk) => {
+          return {
+            KV: {
+              Verb: 'delete-cas',
+              Key: rk.KV.Key,
+              Index: rk.KV.ModifyIndex,
+            },
+          }
+        })
+        let deleteResp
+        try {
+          deleteResp = await cc.transaction.create(deleteTxn)
+        } catch (e) {
+          if (e.statusCode !== 409) {
+            console.error(`Error deleting keys: `, e)
+          }
+        }
+        if (deleteResp && !deleteResp.Errors) {
+          cleaned++
+          console.log(`cleaned routes for ${key}`)
+        }
+      }
+      const msg = `${i}/${cleaned}/${toClean.length}`
+      res.write(msg + '\n')
+      i++
+    }
+    res.end(`cleaned routes for ${cleaned}/${toClean.length} streams\n`)
+  },
+)
+
+app.post(
+  '/active-clean',
+  authMiddleware({ anyAdmin: true }),
+  async (req, res) => {
+    // check if there is streams marked 'active' but wasn't seen recently
+
+    const query = []
+    query.push(sql`data->>'isActive' = 'true'`)
+
+    let docs, cursor
+    const now = Date.now()
+    const toClean = []
+    res.writeHead(200)
+    res.flushHeaders()
+    do {
+      ;[docs, cursor] = await db.stream.find(query, { cursor, limit: 100 })
+      // sending progress should prevent request timing out
+      res.write('.')
+      for (const stream of docs) {
+        const lastSeen = new Date(stream.lastSeen).getTime()
+        const needClean = isNaN(lastSeen) || now - lastSeen > ACTIVE_TIMEOUT
+        if (needClean) {
+          toClean.push(stream)
+        }
+      }
+    } while (cursor && docs.length === 100)
```

Comment:
No, it isn't
```js
    return [docs, docs[docs.length - 1].id]
```
Even if there is only one document present, it will return id of the last document.

Reply:
Ah, boo. Should probably change that but alright.

Reply:
There is fix for that in my other PR )